packages/hint-no-broken-links/src/hint.ts

```diff
@@ -169,15 +146,56 @@ export default class NoBrokenLinksHint implements IHint {
             fetchedUrls.push({ statusCode: fetchEnd.response.statusCode, url: fetchEnd.resource });
         };
 
-        context.on('element::img', validateElementSrcs);
-        context.on('element::a', validateElementSrcs);
-        context.on('element::audio', validateElementSrcs);
-        context.on('element::video', validateElementSrcs);
-        context.on('element::link', validateElementSrcs);
-        context.on('element::script', validateElementSrcs);
-        context.on('element::source', validateElementSrcs);
-        context.on('element::track', validateElementSrcs);
-        context.on('element::object', validateElementSrcs);
+        const validateCollectedUrls = async (event: TraverseEnd) => {
+            const { resource } = event;
+            const pageDOM: IAsyncHTMLDocument = context.pageDOM as IAsyncHTMLDocument;
+            const baseTags: Array<IAsyncHTMLElement> = await pageDOM.querySelectorAll('base');
+            const hrefAttribute = (baseTags.length === 0) ? null : baseTags[0].getAttribute('href');
+
+            const reports: Array<Promise<void>> = collectedElementsWithUrls.reduce<Promise<void>[]>((accumulatedReports, [element, urls]) => {
+                return [...accumulatedReports, ...urls.map(async (url) => {
+                    const isRelativeUrl = (!url.startsWith('/') && URL.parse(url).hostname === null);
```

Comment:
I believe this is part of the [Legacy URL API](https://nodejs.org/dist/latest-v8.x/docs/api/url.html#url_legacy_url_api). It seems we didn't catch this on the original PR ☹

Reply:
I changed the code now to use the new WHATWG URL API instead. I initially thought I should better keep it in the way it was written before.

Reply:
And you were right on doing that. On the original PR for this rule we didn't realize we were using the old API. Sorry about that and thanks for updating it!